src/commands/createRepo/RepoCreateStep.ts

```diff
@@ -22,29 +24,46 @@ export class RepoCreateStep extends AzureWizardExecuteStep<IStaticWebAppWizardCo
+            localize('creatingPublicGitHubRepo', 'Creating new public GitHub repository "{0}"...', newRepoName);
         ext.outputChannel.appendLog(creatingGitHubRepo);
         progress.report({ message: creatingGitHubRepo });
 
-        const repo: Repository = nonNullProp(wizardContext, 'repo');
+        const client: Octokit = await createOctokitClient(wizardContext.accessToken);
+        const gitHubRepoRes: RepoCreateData = (isUser(wizardContext.orgData) ? await client.repos.createForAuthenticatedUser({ name: newRepoName, private: newRepoIsPrivate }) :
+            await client.repos.createInOrg({ org: nonNullProp(wizardContext, 'orgData').login, name: newRepoName, private: newRepoIsPrivate })).data;
+        wizardContext.repoHtmlUrl = gitHubRepoRes.html_url;
+
+        const git: API = await getGitApi();
+        const fsPath: string = nonNullProp(wizardContext, 'fsPath');
+        const uri: Uri = Uri.file(fsPath);
+        let repo: Repository | null = git.getRepository(uri);
+
+        if (!repo) {
+            // if there is no repo, it needs to be initialized
+            // https://github.com/microsoft/vscode/issues/111210
+            repo = nonNullValue(await git.init(uri));
+            ext.outputChannel.appendLog(localize('initRepo', 'Initialized repository in local workspace "{0}".', basename(fsPath)));
+        }
+
         if (!repo.state.HEAD?.commit) {
             // needs to have an initial commit
             await repo.commit(localize('initCommit', 'Initial commit'), { all: true });
+            ext.outputChannel.appendLog(localize('commitRepo', 'Created initial commit in local repository.'));
         }
 
-        const client: Octokit = await createOctokitClient(wizardContext.accessToken);
-        const gitHubRepoRes: RepoCreateData = (isUser(wizardContext.orgData) ? await client.repos.createForAuthenticatedUser({ name: newRepoName, private: newRepoPrivacy }) :
-            await client.repos.createInOrg({ org: nonNullProp(wizardContext, 'orgData').login, name: newRepoName, private: newRepoPrivacy })).data;
-        wizardContext.repoHtmlUrl = gitHubRepoRes.html_url;
-
         const remoteName: string = nonNullProp(wizardContext, 'newRemoteShortname');
         await repo.addRemote(remoteName, gitHubRepoRes.clone_url);
         const branch: Branch = await repo.getBranch('HEAD');
+
+        const pushingBranch: string = localize('pushingBranch', 'Pushing local branch "{0}" to GitHub repository "{1}"...', branch.name, wizardContext.newRepoName);
+        ext.outputChannel.appendLog(pushingBranch);
+        progress.report({ message: pushingBranch });
         await repo.push(remoteName, branch.name, true);
 
-        const createdGitHubRepo: string = localize('createdGitHubRepo', 'Created new {0} GitHub repository "{1}"', privacy, newRepoName);
+        const createdGitHubRepo: string = newRepoIsPrivate ? localize('createdPrivateGitHubRepo', 'Created new private GitHub repository "{0}".', newRepoName) :
```

Comment:
Hmm, well I don't really think the repo is considered "created" (at least from our extension's perspective-- an empty repo will exist) until after the push completes.  Think we should just say push completed then?

Reply:
What do you mean it's not considered "created"? Just because a resource group is empty, doesn't mean we wait to tell users we created it until we fill it with resources

Reply:
Yeah, I see what you mean. I just added a log signifying when the push is complete as well.